pandas/tests/frame/test_constructors.py

```diff
@@ -194,13 +194,11 @@ def test_series_with_name_not_matching_column(self):
         [
             lambda: DataFrame(),
             lambda: DataFrame(None),
-            lambda: DataFrame({}),
             lambda: DataFrame(()),
             lambda: DataFrame([]),
             lambda: DataFrame(_ for _ in []),
             lambda: DataFrame(range(0)),
             lambda: DataFrame(data=None),
-            lambda: DataFrame(data={}),
```

Comment:
Why does an empty dict not produce RangeIndexes?

Reply:
The change in this PR is that if `Index=None` and we can not infer the index from the data, then the index is a `RangeIndex`, also in the zero length case.

IMO dicts should be for empty dicts considered similar to supplying the dict's values and keys separately, and e.g. `Series(mydict.values() index=mydict.keys())` is for empty dicts the most equivalent to `Series([] index=[])`, which means an `Index` with object dtype.

Also notice that `DataFrame({})` after this PR will be considered equivalent to `DataFrame({}.values(), index=None, columns={}.keys())`, so the DataFrame index will now be a `RangeIndex`, while the columns will be a `Index[dtype=object]`. Before this PR both axes were `Index[dtype=object]`.